src/org/mozilla/javascript/NativeArray.java

```diff
@@ -1962,6 +1970,24 @@ private static Object js_copyWithin(
         return thisObj;
     }
 
+    private static Object js_at(Context cx, Scriptable scope, Scriptable thisObj, Object[] args) {
+        int k;
+        Scriptable o = ScriptRuntime.toObject(cx, scope, thisObj);
+        int len = (int) getLengthProperty(cx, o);
```

Comment:
It seems bad to cast to int without inspecting length and index.
```js
var arrayLike = {
  0: 0,
  length: 0x7fffffff + 1  // Integer.MAX_VALUE + 1
};
console.log(Array.prototype.at.call(arrayLike, 0)); // expected 0, but actual undefined
```

Reply:
Thanks for the feedback - I just redeployed and changed NativeArray js_at so that it no longer casts len. I also added argument checks to all functions. I added your example as a test and it passes.

Please let me know if it needs any additional changes.